Source/Core/Gen4/Encounters4.cpp

```diff
@@ -272,6 +272,125 @@ namespace Encounters4
                 }
             }
 
+            if (encounter == Encounter::SafariZone)
+            {
+                size_t size = hgss_safari.size();
+
+                for (size_t offset = 0; offset < size; )
+                {
+                    const u8 *entry = hgss_safari.data() + offset;
+                    int grassBlockSlot = 0;
+                    int waterBlockSlot = 0;
+
+                    u8 location = entry[0];
+                    u8 waterFlag = entry[1];
+
+                    std::vector<Slot> slots;
+                    for (int i = 0; i < 10; i++)
+                    {
+                        u16 specie = *reinterpret_cast<const u16 *>(entry + (30 * modifier) + 2 + (i * 3));
+                        u8 level = entry[(30 * modifier) + 4 + (i * 3)];
+
+                        while (grassBlockSlot < 10)
+                        {
+                            if (blocks[entry[182 + (grassBlockSlot * 4)]] >= entry[183 + (grassBlockSlot * 4)] &&
+                                    blocks[entry[184 + (grassBlockSlot * 4)]] >= entry[185 + (grassBlockSlot * 4)])
+                            {
+                                specie = *reinterpret_cast<const u16 *>(entry + (30 * modifier) + (grassBlockSlot * 3) + 92);
+                                level = entry[(30 * modifier) + (grassBlockSlot * 3) + 94];
+                                grassBlockSlot++;
+                                break;
+                            }
+
+                            grassBlockSlot++;
+                        }
+
+                        // Adjust slot according to Safari encounter type
+                        if (waterFlag != 0 && safariEncounter == Encounter::Surfing)
+                        {
+                            specie = *reinterpret_cast<const u16 *>(entry + (30 * modifier) + (i * 3) + 222);
+                            level = entry[(30 * modifier) + (i * 3) + 224];
+
+                            while (waterBlockSlot < 3)
+                            {
+                                if (blocks[entry[339 + (waterBlockSlot * 4)]] >= entry[340 + (waterBlockSlot * 4)] &&
+                                        blocks[entry[341 + (waterBlockSlot * 4)]] >= entry[342 + (waterBlockSlot * 4)])
+                                {
+                                    specie = *reinterpret_cast<const u16 *>(entry + (9 * modifier) + (waterBlockSlot * 3) + 312);
+                                    level = entry[(9 * modifier) + (waterBlockSlot * 3) + 314];
+                                    waterBlockSlot++;
```

Comment:
This looks unnecessary

Reply:
This changes the slots based on the blocks you insert, so this is 100% necessary. If you have enough blocks it will change the slot